packages/python/accern_data/accern_data.py

```diff
@@ -813,8 +833,18 @@ def iterate_range(
         if end_date is None:
             end_date = start_date
         indicator_obj = self._get_valid_indicator(indicator)
-        indicator_obj.generate_bar(
-            total=len(pd.date_range(start_date, end_date)))
+        try:
+            indicator_obj.generate_bar(
+                total=len(pd.date_range(start_date, end_date)))
+        except ImportError:
```

Comment:
do this inside of valid_indicator

Reply:
```
IS_INDICATOR_BAR_AVAILABLE: Optional[bool] = None

def is_indicator_bar_available() -> bool:
global IS_INDICATOR_BAR_AVAILABLE
if IS_INDICATOR_BAR_AVAILABLE is not None:
   return IS_INDICATOR_BAR_AVAILABLE
try:
    import ...
    IS_INDICATOR_BAR_AVAILABLE = True
    return True
except ImportError:
    IS_INDICATOR_BAR_AVAILABLE = False
    return False
```

Reply:
then use this is... function inside valid indicator to return the message directly and set the warning

Reply:
```
global IS_INDICATOR_BAR_AVAILABLE
if IS_INDICATOR_BAR_AVAILABLE is not None:
   return IS_INDICATOR_BAR_AVAILABLE
```

Above caching won't work in case if we switch from `pbar` to `message`. We would still see warning for `message`.